content/articles/constructors-in-kotlin/index.md

```diff
@@ -0,0 +1,180 @@
+
+- The primary constructor
+- The secondary constructor
+
+#### Primary constructors
+The primary constructor is a section of the class header that comes after the class name. The constructor keyword is used after the class name. The example below exemplifies the definition:
+
+```kotlin
+class student Constructor(name:string,age:Int,Email:string)
+ { 
+     ///code to be processed
+}
+```
+The constructor keyword can be ignored if the primary constructor has no annotations or accessibility modifiers:
+
+In the preceding example, we are not going to use the `constructor` keyword. By default, all the constructors are public which means they are visible everywhere in the class. Therefore we don't have to include the `constructor` keyword, as shown below:
+
+```kotlin
+class student(name:string,age:Int,Email:string){ }
+```
+
+#### Secondary constructor
+In Kotlin, a secondary constructor is created using the `constructor` keyword. In addition, the secondary constructor can be used to call the primary constructor. The example below shows how you can declare two constructors with a student having two parameters (name and year). 
+
+```kotlin
+class student{ 
+    constructor(name: Int){ 
+
+    } 
+    constructor(year:Int, id: Int){ 
+
+    } 
+}
+```
+
+Also, the secondary constructor can assign values to fields while the class is instantiated. The example below illustrates this property:
+
+```kotlin
+class myStudent{ 
+    constructor(name:String,id:Int{
+         println("Name=${name}") 
+         println("Id=${id}")
+    } 
+         
+    fun main(args: Array <string>) { 
+        val mystudent=myClass("Alex",200) 
+    }
+}
+```
+
+The output of the above code will look like this:
+
+```
+Name = Alex 
+Id = 200
+```
+
+Kotlin also offers room for using a primary constructor with a secondary constructor in the same class. You need to use the `this` keyword for authorization:
+
+```kotlin
+class myStudent(email:String){ 
+    constructor(name:String,id:Int,email:string):this(email){
+         println("Name = ${name}") 
+         println("Email= ${email}") 
+    } 
+    
+    fun main(args:Array<string>){
+         val mystudent = myStudent("Alex","alex@gmail") 
+    }      
+}
+```
+
+This outputs:
+
+```
+Name = Alex Email = Alex@gmail
+```
+
+Constructors can also be called by another secondary constructor of the same class:
+
+```kotlin
+class myStudent{
+     constructor(name:String,id:Int): this(name,id,"Alex@gmail"){
+        println("Name=${name}")
+         println("Id=${id}")
+    }
+    constructor(name:string, id:Int, mail:string ) 
+      { 
+          println("Name = ${name}") 
+          println("Id" = ${id})
+           println("Email = ${mail}")
+     }
+}    
+     fun main(args:Array<string>){
+     val mystudent = myStudent("Alex",200) 
+    
+}
+```
+The output of the above code is:
+
+```
+Name = Alex 
+Id = 200
+Email = Alex@gmail
+```
+
+### Init blocks

```

Comment:
You never properly explained when to use the init block despite saying you would in the intro. It sounds vague as to why you would use this over a secondary constructor.